tests/e2e/csi_snapshot_basic.go

```diff
@@ -1355,4 +1356,615 @@ var _ = ginkgo.Describe("[block-vanilla-snapshot] Volume Snapshot Basic Test", f
 		err = verifySnapshotIsDeletedInCNS(volHandle, snapshotId)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+
+	/*
+		Volume deletion with existing snapshots
+		1. Create a sc, and a pvc using this sc
+		2. Create a dynamic snapshot using above pvc as source
+		3. Delete this pvc, expect the pvc to be deleted successfully
+		4. Underlying pv should not be deleted and should have a valid error
+		    calling out that the volume has active snapshots
+			(note: the storageclass here is set to Delete retentionPolicy)
+		5. Expect VolumeFailedDelete error with an appropriate err-msg
+		6. Run cleanup - delete the snapshots and then delete pv
+	*/
+	ginkgo.It("Volume deletion with existing snapshots", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var pvclaims []*v1.PersistentVolumeClaim
+		var err error
+		var snapshotContentCreated = false
+		var isPVCDeleted = false
+		var isPVDeleted = false
+
+		ginkgo.By("Create storage class and PVC")
+		scParameters[scParamDatastoreURL] = datastoreURL
+		storageclass, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client,
+			pvclaim.Namespace, pvclaim.Name, framework.Poll, time.Minute)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+		pvclaims = append(pvclaims, pvclaim)
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client, pvclaims, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			if !isPVCDeleted {
+				err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				isPVCDeleted = true
+			}
+			if !isPVDeleted {
+				err := client.CoreV1().PersistentVolumes().Delete(ctx, persistentvolumes[0].Name,
```

Comment:
you can use `fpv.DeletePersistentVolume()` here instead which already ignores `IsNotFound` error

Reply:
fixed